File: harshadnumbers/harshadnumbers.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

harshad = (n) => {
  digits = String(n).split('').map(d => parseInt(d));
  sd = digits.reduce((a, b) => a + b, 0);
  return n % sd === 0;
}

rl.on('line', (line) => {
  n = parseInt(line);
  while (!harshad(n)) {
    n++;
  }
  console.log(n);
});

File: solutions/harshadnumbers/harshadnumbers.rb
def harshad(n)
  digits = n.to_s.chars.map &:to_i
  sd = digits.inject(0, :+)
  n % sd == 0
end

n = STDIN.readline.to_i
until harshad(n)
  n += 1
end
puts n
